astroid/context.py

```diff
@@ -64,24 +67,20 @@ def __init__(self, path=None, nodes_inferred=None):
         """
         self.callcontext: CallContext | None = None
         """The call arguments and keywords for the given context."""
-        self.boundnode = None
+        self.boundnode: NodeNG | None = None
         """
-        :type: optional[NodeNG]
-
         The bound node of the given context
 
         e.g. the bound node of object.__new__(cls) is the object node
         """
-        self.extra_context = {}
+        self.extra_context: dict[Any, InferenceContext] = {}
```

Comment:
Let's not use `Any`. Since `mypy` isn't running anyway the typing also doesn't give us much so it might be better to just keep the docstring for now until we can be 100% sure we get this correctly.

Reply:
I'd like to keep this one because for some reason Mypy specifically calls it out:

```
astroid/context.py:78: error: Need type annotation for "extra_context" (hint: "extra_context: Dict[<type>, <type>] = ...")  [var-annotated]
```

So when trying to do more typing, it adds noise.